src/classes/RD2_CustomFieldMapper_TEST.cls

```diff
@@ -0,0 +1,289 @@
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Recurring Donations
+* @description Unit tests for the RD -> Opportunity custom field mapper
+*
+*/
+@IsTest(IsParallel=true)
+public class RD2_CustomFieldMapper_TEST {
+
+    private static final Date START_DATE = Date.newInstance(2019, 9, 15);
+    private static final String PAYMENT_CHECK = 'Check';
+    private static final Decimal RD_AMOUNT = 100;
+
+    /***
+     * @description Verifies the Recurring Donation is identified as unchanged
+     * when the record has changed but the field is not part of the custom mapping settings
+     */
+    @IsTest
+    private static void shouldIndicateRDIsUnchangedWhenFieldIsNotInCustomMapping() {
+        createCustomFieldMapping('Name', 'Description');
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder().build();
+
+        npe03__Recurring_Donation__c updatedRD = rd.clone();
+        updatedRD.npe03__Amount__c += 10;
+
+        RD2_CustomFieldMapper fieldMapper = new RD2_CustomFieldMapper();
+        System.assertEquals(false, fieldMapper.hasKeyFieldChanged(updatedRd, rd),
+            'Record should be identified as unchanged since the npe03__Amount__c is not included in the custom mapping');
+    }
+
+    /***
+     * @description Verifies the Recurring Donation is identified as changed
+     * when the record has changed and the field is part of the custom mapping settings
+     */
+    @IsTest
+    private static void shouldIndicateRDIsChangedWhenFieldIsInCustomMapping() {
+        createCustomFieldMapping('npe03__Amount__c', 'Amount');
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder().build();
+
+        npe03__Recurring_Donation__c updatedRD = rd.clone();
+        updatedRD.npe03__Amount__c += 10;
+
+        RD2_CustomFieldMapper fieldMapper = new RD2_CustomFieldMapper();
+        System.assertEquals(true, fieldMapper.hasKeyFieldChanged(updatedRd, rd),
+            'Record should be identified as changed since the npe03__Amount__c is included in the custom mapping');
+    }
+
+    /***
+     * @description Verifies an Opportunity is unchanged when custom field mapping is empty
+     */
+    @IsTest
+    private static void shouldNotChangeOpportunityWhenCustomMappingIsEmpty() {
+        delete [SELECT Id FROM npe03__Custom_Field_Mapping__c];
```

Comment:
I'm not sure why this is necessary. Like all other data, all objects start as empty in a unit test.